src/elements/e-video.vue

```diff
@@ -97,108 +85,126 @@
           // eslint-disable-next-line no-console
           console.error("Neither 'videoId' or 'videoUrl' were defined", this.$el);
 
           return undefined;
         }
 
         if (this.videoUrl && !this.videoId) {
-          if (!this.validateUrl()) {
+          if (!this.validateUrl) {
             return undefined;
           }
 
-          if (this.source === 'youtube') {
-            /**
-             * Extract id from video url
-             */
-            const id = this.videoUrl.substring(
-              this.videoUrl.indexOf('v=') + 2,
-              this.videoUrl.indexOf('&')
-            );
+          let id: string | null;
 
-            return `${playerBaseUrl}${id}`;
-          }
+          switch (this.source) {
+            case VideoSource.Youtube:
+              id = new URLSearchParams(new URL(this.videoUrl).search).get('v');
 
-          const id = this.videoUrl.substring(
-            this.videoUrl.indexOf(videoUrlSources[this.source])
-              + videoUrlSources[this.source].length,
-            this.videoUrl.length
-          );
+              return this.replaceIdInUrl(playerBaseUrl, id);
 
-          return `${playerBaseUrl}${id}`;
+            case VideoSource.Vimeo:
+              id = new URL(this.videoUrl).pathname.slice(1);
+
+              return this.replaceIdInUrl(playerBaseUrl, id);
+
+            default:
+              return undefined;
+          }
         }
 
-        if (this.videoId.includes(videoUrlSources[this.source])) {
+        /**
+         * The videoId can consist of lowercase/uppercase letters, numbers, dashes and underscores.
+         */
+        if ((/[a-zA-Z0-9-_]/).test(this.videoId) === false) {
           // eslint-disable-next-line no-console
           console.error('video-id can not be part of a URL');
 
           return undefined;
         }
 
-        return `${playerBaseUrl}${this.videoId}`;
+        return this.replaceIdInUrl(playerBaseUrl, this.videoId);
       },
-    },
 
-    // computed: {},
-    watch: {
       /**
-       * This should check whether responsive or height + width are specified
+       * Returns
        */
-      $props: {
-        immediate: true,
-        handler() {
-          if (this.responsive && (this.width || this.height)) {
-            console.error('Either use responsive or height and width');
-          } else if (
-            (this.width && !this.height)
-            || (!this.width && this.height)
-          ) {
-            console.error('width and height both need to be specified');
-          }
-        },
-      },
-      videoUrl: {
-        immediate: true,
-        handler() {
-          this.validateUrl();
-        },
+      hasFixedSize(): boolean {
+        return !this.width && !this.height;
       },
     },
 
+    // computed: {},
+    // watch: {},
+
     // beforeCreate() {},
-    // created() {},
+    created() {
+      if (this.width || this.height) {
+        if (!this.height && !this.width) {
```

Comment:
todo: I don't think this condition makes sense. If the first one passed, the second one can never be true...?